"Null pointer dereference" error on 0.3.0

Happens on pod startup on `ubuntu-16-04-x64`.

Pod log: 
```
ts=2017-05-19T17:01:48Z caller=main.go:77 component=platform host=https://10.96.0.1:443
ts=2017-05-19T17:01:48Z caller=main.go:89 component=platform kubectl=/usr/local/bin/kubectl
ts=2017-05-19T17:01:48Z caller=main.go:101 component=platform services=4
ts=2017-05-19T17:01:48Z caller=daemon.go:91 component=client connecting=true
ts=2017-05-19T17:01:48Z caller=main.go:134 addr=:3031
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xac291b]

goroutine 43 [running]:
panic(0x136d600, 0xc420014020)
	/usr/local/go/src/runtime/panic.go:500 +0x1a1
github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt.safeError.func1(0x1d27740, 0xc420385580, 0xc4204a7be8, 0xc4204a7bf8)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt/encode.go:272 +0xf6
panic(0x136d600, 0xc420014020)
	/usr/local/go/src/runtime/panic.go:458 +0x243
github.com/weaveworks/flux/http/websocket.DialErr.Error(0xc4202ee980, 0x0, 0x1daa818, 0x0)
	/home/ubuntu/src/github.com/weaveworks/flux/http/websocket/client.go:21 +0x6b
github.com/weaveworks/flux/http/websocket.(*DialErr).Error(0xc4203dfdf0, 0x1dacce0, 0xc4204a7b88)
	<autogenerated>:1 +0x5b
github.com/weaveworks/flux/vendor/github.com/pkg/errors.(*withMessage).Error(0xc420385540, 0xc4204a7b98, 0xc4204a7ba8)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/pkg/errors/errors.go:195 +0x37
github.com/weaveworks/flux/vendor/github.com/pkg/errors.(*withStack).Error(0xc420385580, 0x158f4e8, 0x1d27740)
	<autogenerated>:8 +0x49
github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt.safeError(0x1d27740, 0xc420385580, 0x0, 0x0, 0x14d9200)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt/encode.go:276 +0x99
github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt.writeValue(0x1d26d00, 0xc420070910, 0x13a7820, 0xc420385580, 0x1, 0x0)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt/encode.go:205 +0x1f8
github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt.(*Encoder).EncodeKeyval(0xc420070900, 0x12e7da0, 0xc4203b6830, 0x13a7820, 0xc420385580, 0x0, 0x0)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt/encode.go:60 +0x102
github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt.(*Encoder).EncodeKeyvals(0xc420070900, 0xc4203d9440, 0x8, 0xc, 0xc42028f600, 0xc4204a7dc0)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-logfmt/logfmt/encode.go:83 +0x11c
github.com/weaveworks/flux/vendor/github.com/go-kit/kit/log.logfmtLogger.Log(0x1d2c580, 0xc420022018, 0xc4203d9440, 0x8, 0xc, 0x0, 0x0)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-kit/kit/log/logfmt_logger.go:46 +0xdf
github.com/weaveworks/flux/vendor/github.com/go-kit/kit/log.(*logfmtLogger).Log(0xc420423300, 0xc4203d9440, 0x8, 0xc, 0xc4203855a0, 0x2)
	<autogenerated>:11 +0x79
github.com/weaveworks/flux/vendor/github.com/go-kit/kit/log.(*Context).Log(0xc420306ab0, 0xc4203855a0, 0x2, 0x2, 0xc4203b6830, 0x0)
	/home/ubuntu/src/github.com/weaveworks/flux/vendor/github.com/go-kit/kit/log/log.go:86 +0x182
github.com/weaveworks/flux/http.(*Daemon).loop(0xc4202eec80)
	/home/ubuntu/src/github.com/weaveworks/flux/http/daemon.go:75 +0x20f
created by github.com/weaveworks/flux/http.NewDaemon
	/home/ubuntu/src/github.com/weaveworks/flux/http/daemon.go:61 +0x27c
```

kubctl version:
```
Server Version: version.Info{Major:"1", Minor:"6", GitVersion:"v1.6.1", GitCommit:"b0b7a323cc5a4a2019b2e9520c21c7830b7f708e", GitTreeState:"clean", BuildDate:"2017-04-03T20:33:27Z", GoVersion:"go1.7.5", Compiler:"gc", Platform:"linux/amd64"}
```

k8s deployment spec:
```
    spec:
      template:
        metadata:
          namespace: kube-system
          labels:
            name: weave-flux-agent
            app: weave-flux
            weave-cloud-component: flux
            weave-flux-component: agent
        spec:
          containers:
            - name: agent
              image: 'quay.io/weaveworks/fluxd:0.3.0'
              imagePullPolicy: IfNotPresent
              args:
                - '--token=<REDACTED>'
          serviceAccountName: weave-flux
      replicas: 1
    apiVersion: apps/v1beta1
    kind: Deployment
```

--- a/http/websocket/client.go
+++ b/http/websocket/client.go
@@ -18,7 +18,10 @@ type DialErr struct {
 }
 
 func (de DialErr) Error() string {
-	return fmt.Sprintf("connecting websocket %s (http status code = %v)", de.URL, de.HTTPResponse.StatusCode)
+	if de.URL != nil && de.HTTPResponse != nil {
+		return fmt.Sprintf("connecting to websocket %s (http status code = %v)", de.URL, de.HTTPResponse.StatusCode)
+	}
+	return "connecting to websocket (unknown error)"
 }
 
 // Dial initiates a new websocket connection.

--- a/http/websocket/client.go
+++ b/http/websocket/client.go
@@ -41,7 +41,10 @@ func Dial(client *http.Client, ua string, token flux.Token, u *url.URL) (Websock
 	// Use http client to do the http request
 	conn, resp, err := dialer(client).Dial(u.String(), req.Header)
 	if err != nil {
-		return nil, &DialErr{u, resp}
+		if resp != nil {
+			err = &DialErr{u, resp}
+		}
+		return nil, err
 	}
 
 	// Set up the ping heartbeat

